sdk/webpubsub/Azure.Messaging.WebPubSub/README.md

```diff
@@ -0,0 +1,121 @@
+
+Any scenario that requires real-time publish-subscribe messaging between server and clients or among clients, can use Azure Web PubSub service. Traditional real-time features that often require polling from server or submitting HTTP requests, can also use Azure Web PubSub service.
+
+This library can be used to do the following actions. Details about the terms used here are described in [Key concepts](#terms) section.
+
+- Send messages to hubs and groups. 
+- Send messages to particular users and connections.
+- Organize users and connections into groups.
+- Close connections
+- Grant, revoke, and check permissions for an existing connection
+
+[Source code](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/webpubsub/Azure.Messaging.WebPubSub/src) |
+[Package (NuGet)](https://www.nuget.org/packages/Azure.Messaging.WebPubSub/) |
+[API reference documentation](https://aka.ms/awps/sdk/csharp) |
+[Product documentation](https://aka.ms/awps/doc) |
+[Samples][samples_ref]
+
+## Getting started
+
+### Install the package
+
+Install the client library from [NuGet](https://www.nuget.org/):
+
+```PowerShell
+dotnet add package Azure.Messaging.WebPubSub --prerelease
+```
+
+### Prerequisites
+
+- An [Azure subscription][azure_sub].
+- An existing Azure Web PubSub service instance.
+
+### Create a `WebPubSubServiceClient`
+
+```csharp
+var serviceClient = new WebPubSubServiceClient(new Uri("<endpoint>"), "<hub>", new AzureKeyCredential("<access-key>"));
+```
+
+
+## Key concepts
+
+### Connection
+
+Connections, represented by a connection id, represent an individual websocket connection to the Web PubSub service. Connection id is always unique.
+
+### Hub
+
+Hub is a logical concept for a set of connections. Connections are always connected to a specific hub. Messages that are broadcast to the hub are dispatched to all connections to that hub. Hub can be used for different applications, different applications can share one Azure Web PubSub service by using different hub names.
+
+### Group
+
+Group allow broadcast messages to a subset of connections to the hub. You can add and remove users and connections as needed. A client can join multiple groups, and a group can contain multiple clients.
+
+### User
+
+Connections to Web PubSub can belong to one user. A user might have multiple connections, for example when a single user is connected across multiple devices or multiple browser tabs.
+
+### Message
+
+Using this library, you can send messages to the client connections. A message can either be string text, JSON or binary payload.
+
+## Examples
+
+### Broadcast a text message to all clients
+
+```csharp
```

Comment:
Consider using snippets that are injected from tests,  https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#updating-sample-snippets

Reply:
will do after the first beta. I will capture these in an issue.